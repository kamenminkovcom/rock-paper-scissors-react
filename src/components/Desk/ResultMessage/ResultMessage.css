.result-message {
    background-image: none;
    border: 2px solid;
    padding: .375rem .75rem;
    font-size: 1rem;
    line-height: 1.5;
    border-radius: .25rem;
    margin: 10%;
    text-align: center;
}