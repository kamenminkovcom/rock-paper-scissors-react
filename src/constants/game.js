export const handTypes = {
    paper: 'paper',
    rock: 'rock',
    scissors: 'scissors',
};

export const gameRules = {
    [handTypes.paper]: {
        [handTypes.paper]: 0,
        [handTypes.rock]: 1,
        [handTypes.scissors]: 0,
    },
    [handTypes.rock]: {
        [handTypes.rock]: 0,
        [handTypes.scissors]: 1,
        [handTypes.paper]: 0
    },
    [handTypes.scissors]: {
        [handTypes.scissors]: 0,
        [handTypes.paper]: 1,
        [handTypes.rock]: 0
    }
};

export const possibleResults = {
    win: 'win',
    lose: 'lose',
    draw: 'draw'
};

//may we would like to change classNames and it should not break the logic
export const classNames = {
    [possibleResults.win]: 'win',
    [possibleResults.lose]: 'lose',
    [possibleResults.draw]: 'draw'
};

export const gameHistoryCount = 5;

export const defaultChoices = {
    cpuChoice: undefined,
    playerChoice: undefined
};