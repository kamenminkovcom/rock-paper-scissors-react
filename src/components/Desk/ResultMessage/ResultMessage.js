import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import withCondition from '../../../HOC/WithCondition';
import {classNames, possibleResults} from '../../../constants/game';

import './ResultMessage.css';

const ResultMessage = ({message, result}) => (
    <div className={classnames(classNames[result],
        'result-message',
        `${classNames[result]}-border`,
        'uppercase')}>
        {message}
    </div>
);

//may we will change the messages, so we are receiving message and result props
//for now they are equal
ResultMessage.propTypes = {
    message: PropTypes.string.isRequired,
    result: PropTypes.oneOf(Object.values(possibleResults))
};

export default withCondition(ResultMessage);