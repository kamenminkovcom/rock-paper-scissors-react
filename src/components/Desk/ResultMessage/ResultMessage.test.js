import React from 'react';
import {shallow} from 'enzyme';
import ResultMessage from './ResultMessage';
import {possibleResults} from '../../../constants/game';

describe('<ResultMessage/>', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<ResultMessage message={possibleResults.win} result={possibleResults.win}/>);
        expect(wrapper).toMatchSnapshot();
    })
});