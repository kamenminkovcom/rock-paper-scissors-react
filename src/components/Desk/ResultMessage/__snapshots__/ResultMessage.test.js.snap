// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultMessage/> Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultMessage
    message="win"
    result="win"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "win",
      "className": "win result-message win-border uppercase",
    },
    "ref": null,
    "rendered": "win",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "win",
        "className": "win result-message win-border uppercase",
      },
      "ref": null,
      "rendered": "win",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
